stl/src/filesys.cpp

```diff
@@ -351,10 +350,10 @@ _FS_DLL int __CLRCALL_PURE_OR_CDECL _Equivalent(
     } else if (!_Ok1 || !_Ok2) {
         return 0;
     } else { // test existing files for equivalence
-        return _Info1.VolumeSerialNumber != _Info2.VolumeSerialNumber
-                    || memcmp(&_Info1.FileId, &_Info2.FileId, sizeof(_Info1.FileId)) != 0
-                 ? 0
-                 : 1;
+        return _Info1.VolumeSerialNumber == _Info2.VolumeSerialNumber
+                    && memcmp(&_Info1.FileId, &_Info2.FileId, sizeof(_Info1.FileId)) == 0
+                 ? 1
+                 : 0;

```

Comment:
I don't see why we can't compare `FILE_ID_INFO` as a whole thing with a single `memcmp`